ciao-controller/command.go

```diff
@@ -106,12 +106,25 @@ func (c *controller) confirmTenantRaw(tenantID string) error {
 	}
 
 	if tenant == nil {
-		tenant, err = c.ds.AddTenant(tenantID)
+		_, err = c.ds.AddTenant(tenantID)
 		if err != nil {
 			return err
 		}
 	}
 
+	return nil
+}
+
+func (c *controller) confirmCNCI(tenantID string) error {
```

Comment:
I endorse starting the cnci only on instance add. 

My concern about the raciness (that the code in confirmTenant()) tried to solve. I think if you adjust the way you split confirmCNCI() out from confirmTenantRaw() then you'd get that (renaming confirmTenant() to confirmCNCI() and then creating a confirmTenant() that only does the adding of the tenant.)